### PR TITLE
move segment existency check to preprocess

### PR DESCRIPTION
**Pull request by zhtaoxiang**
Similar to `preUploadSegments` and `postUploadSegments`, we should not put lineage related logic directly into `executeTask`, instead, we should put it into `preprocess` so that child classes can override it.

### CODE REVIEW

File: pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/BaseMultipleSegmentsConversionExecutor.java [hunk 1]
```diff
@@ -107,14 +107,31 @@ protected abstract List<SegmentConversionResult> convert(PinotTaskConfig pinotTa
...
+          tableNameWithType, nonExistingSegmentNames));
+    }
   }
 
   /**
    * Post processing operations to be done before exiting a successful task execution
    */
-  protected void postProcess(PinotTaskConfig pinotTaskConfig) {
+  protected void postProcess(PinotTaskConfig pinotTaskConfig) throws Exception {
   }
```
Comment (walterddr) [inline pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/BaseMultipleSegmentsConversionExecutor.java:135]: is this change needed?

**Review by walterddr — APPROVED** [review]
lgtm

File: pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/BaseMultipleSegmentsConversionExecutor.java (hunk 1 above)
Comment (zhtaoxiang) [reply pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/BaseMultipleSegmentsConversionExecutor.java:135]: The purpose is to make it consistent with `preUploadSegmentsm`,  `postUploadSegments` and `preprocess`

**Review by snleee — APPROVED** [review]
I think that we should do sth like the following in the long run:

1. provide preprocess/preUpload/postUpload as today
2. provide an option to set consistent push protocol (since this is a widely used building block)

In this way, the developers can just worry about adding some extra metadata that they need to modify.

In today, people need to copy the parts updating segment lineage over and over again.
